Lib/test/test_ssl.py

```diff
@@ -2800,6 +2813,39 @@ def test_echo(self):
                 'Cannot create a client socket with a PROTOCOL_TLS_SERVER context',
                 str(e.exception))
 
+    @unittest.skipUnless(support.Py_GIL_DISABLED, "test is only useful if the GIL is disabled")
+    def test_ssl_in_multiple_threads(self):
+        # See GH-124984
+        threads = []
+
+        global USE_SAME_TEST_CONTEXT
+        USE_SAME_TEST_CONTEXT = True
+        try:
+            for func in (
+                self.test_echo,
+                self.test_alpn_protocols,
+                self.test_getpeercert,
+                self.test_crl_check,
+                self.test_check_hostname_idn,
+                self.test_wrong_cert_tls12,
+                self.test_wrong_cert_tls13,
+            ):
+                # Be careful with the number of threads here.
+                # Too many can result in failing tests.
+                for num in range(5):
+                    with self.subTest(func=func, num=num):
+                        threads.append(Thread(target=func))
+
+            for thread in threads:
+                with self.subTest(thread=thread):
+                    thread.start()
+
+            for thread in threads:
+                with self.subTest(thread=thread):
+                    thread.join()
```

Comment:
I'm not sure that calling start()+join() reports properly errors to this thread/function.

You should use something like `with catch_threading_exception() as cm:` of `test.support.threading_helper` and check `cm.exc_value`.